mesonpy/__init__.py

```diff
@@ -542,6 +553,60 @@ def build(self, directory: Path) -> pathlib.Path:
 
         return wheel_file
 
+    def build_editable(self, directory: Path) -> pathlib.Path:
+        self._project.build()  # ensure project is built
+
+        wheel_file = pathlib.Path(directory, f'{self.name}.whl')
+
+        install_path = self._source_dir / '.mesonpy' / 'editable' / 'install'
+        import_paths = {
+            os.fspath(install_path.joinpath(
+                *self._sysconfig_paths[path].split(os.path.sep)
+            ))
+            for path in ('purelib', 'platlib')
+        }
+        top_level_modules = self._project.top_level_modules
+        rebuild_commands = [list(cmd) for cmd in self._project.build_commands(install_path)]
+
+        install_path.mkdir(parents=True, exist_ok=True)
+
+        with mesonpy._wheelfile.WheelFile(wheel_file, 'w') as whl:
+            self._wheel_write_metadata(whl)
+            whl.writestr(
+                f'{self.distinfo_dir}/direct_url.json.py',
+                f'file://{os.fspath(self._source_dir)}'.encode(),
+            )
+
+            # install hook module
+            hook_module_name = f'_mesonpy_hook_{self.normalized_name.replace(".", "_")}'
+            hook_install_code = textwrap.dedent(f'''
+                MesonpyFinder.install(
+                    project_path='{self._source_dir}',
+                    import_paths={list(import_paths)},
+                    top_level_modules={list(top_level_modules)},
+                    rebuild_commands={rebuild_commands},
+                )
+            ''').strip().encode()
+            whl.writestr(
+                f'{hook_module_name}.py',
+                (importlib_resources.files('mesonpy') / '_editable.py').read_bytes() + hook_install_code,
+            )
+            # install .pth file
+            whl.writestr(
+                f'{self.normalized_name}-editable-hook.pth',
+                f'import {hook_module_name}'.encode(),
+            )
+
+            # install non-code schemes
+            for scheme in self._SCHEME_MAP:
+                if scheme in ('purelib', 'platlib', 'mesonpy-libs'):
+                    continue
+                for destination, origin in self._wheel_files[scheme]:
+                    destination = pathlib.Path(self.data_dir, scheme, destination)
+                    whl.write(origin, destination.as_posix())
+
+        return wheel_file
```

Comment:
Didn't this used to add `meson-python==...` somewhere? I'm not seeing it now, was going to comment on that.

Reply:
Yep, since we don't have any dependencies on the `_editables` module (internal or external), I changed it hook to just copy its contents and append the custom code.